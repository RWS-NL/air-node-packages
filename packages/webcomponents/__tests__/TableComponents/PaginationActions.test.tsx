import { IconButton } from '@material-ui/core';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import TablePaginationActions from '../../src/TableComponents/PaginationActions/PaginationActions';

const onChangePage = jest.fn();

describe('Component Tests', () => {
  let tablePaginationActions: ShallowWrapper;

  beforeAll(() => {
    tablePaginationActions = shallow(
      <TablePaginationActions rowsPerPage={10} page={0} count={42}
        onChangePage={onChangePage} />
    );
  });

  test('should jump to first page', () => {
    const goToFirstPageButton = tablePaginationActions.find(IconButton).first();

    goToFirstPageButton.simulate('click');
    expect(onChangePage).toHaveBeenCalledWith(undefined, 0);
  });

  test('should jump to last page', () => {
    const goToLastPageButton = tablePaginationActions.find(IconButton).last();

    goToLastPageButton.simulate('click');
    expect(onChangePage).toHaveBeenCalledWith(undefined, 4);
  });

  test('should jump to next page', () => {
    const goToNextPageButton = tablePaginationActions.find(IconButton).at(2);

    goToNextPageButton.simulate('click');
    expect(onChangePage).toHaveBeenCalledWith(undefined, 1);
  });

  test('should jump to previous page', () => {
    const goToPreviousPageButton = tablePaginationActions.find(IconButton).at(1);

    goToPreviousPageButton.simulate('click');
    expect(onChangePage).toHaveBeenCalledWith(undefined, -1);
  });
});

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const tablePaginationActions = shallow(
      <TablePaginationActions
        rowsPerPage={10} page={0} count={42}
        onChangePage={onChangePage}
      />
    );
    expect(tablePaginationActions).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const tablePaginationActions = shallow(
      <TablePaginationActions
        rowsPerPage={10} page={0} count={42}
        onChangePage={onChangePage} customclasses={[ 'snapshot-class' ]}
      />
    );
    expect(tablePaginationActions).toMatchSnapshot();
  });
});