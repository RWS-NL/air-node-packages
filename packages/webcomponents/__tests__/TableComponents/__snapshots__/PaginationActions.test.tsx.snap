// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Optional Props 1`] = `
<Styled(MuiBox)
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <Memo()
    data-qa="first-page-button"
    disabled={true}
    icon={<Memo(FirstPageIcon) />}
    onClick={[Function]}
  />
  <Memo()
    data-qa="previous-page-button"
    disabled={true}
    icon={<Memo(KeyboardArrowLeftIcon) />}
    onClick={[Function]}
  />
  <If
    condition={false}
  >
    <Then>
      <WithStyles(ForwardRef(Chip))
        className="activePageChip ie11ChipCorrection"
        color="primary"
        key="1"
        label={1}
        onClick={[Function]}
        variant="default"
      />
    </Then>
    <Else>
      <WithStyles(ForwardRef(Chip))
        className="activePageChip ie11ChipCorrection"
        color="primary"
        key="0"
        label={1}
        onClick={[Function]}
        variant="default"
      />
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="1"
        onClick={[Function]}
      >
        2
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="2"
        onClick={[Function]}
      >
        3
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="3"
        onClick={[Function]}
      >
        4
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="4"
        onClick={[Function]}
      >
        5
      </Styled(MuiBox)>
    </Else>
  </If>
  <Memo()
    data-qa="next-page-button"
    disabled={false}
    icon={<Memo(KeyboardArrowRightIcon) />}
    onClick={[Function]}
  />
  <Memo()
    data-qa="last-page-button"
    disabled={false}
    icon={<Memo(LastPageIcon) />}
    onClick={[Function]}
  />
</Styled(MuiBox)>
`;

exports[`Snapshot Testing Required Props 1`] = `
<Styled(MuiBox)
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <Memo()
    data-qa="first-page-button"
    disabled={true}
    icon={<Memo(FirstPageIcon) />}
    onClick={[Function]}
  />
  <Memo()
    data-qa="previous-page-button"
    disabled={true}
    icon={<Memo(KeyboardArrowLeftIcon) />}
    onClick={[Function]}
  />
  <If
    condition={false}
  >
    <Then>
      <WithStyles(ForwardRef(Chip))
        className="activePageChip ie11ChipCorrection"
        color="primary"
        key="1"
        label={1}
        onClick={[Function]}
        variant="default"
      />
    </Then>
    <Else>
      <WithStyles(ForwardRef(Chip))
        className="activePageChip ie11ChipCorrection"
        color="primary"
        key="0"
        label={1}
        onClick={[Function]}
        variant="default"
      />
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="1"
        onClick={[Function]}
      >
        2
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="2"
        onClick={[Function]}
      >
        3
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="3"
        onClick={[Function]}
      >
        4
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="paginationNumbers"
        component="span"
        key="4"
        onClick={[Function]}
      >
        5
      </Styled(MuiBox)>
    </Else>
  </If>
  <Memo()
    data-qa="next-page-button"
    disabled={false}
    icon={<Memo(KeyboardArrowRightIcon) />}
    onClick={[Function]}
  />
  <Memo()
    data-qa="last-page-button"
    disabled={false}
    icon={<Memo(LastPageIcon) />}
    onClick={[Function]}
  />
</Styled(MuiBox)>
`;
